util.py

```diff
@@ -0,0 +1,12 @@
+def convert_str_to_bool(value: str, default: bool) -> bool:
+    try:
+        if value is None:
+            return default
+        if value.lower() == 'true':
+            return True
+        elif value.lower() == 'false':
+            return False
+        else:
+            return default
+    except Exception as e:
+        return default
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rkj6124/test/pull/2#issuecomment-2399863287>#23c0e7</a><span> - 10/08/2024, 07:09 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The function uses a broad exception catch which may hide unexpected errors and make debugging difficult.
</div>
<div id="fix">
<b>Fix:</b> Catch specific exceptions that are expected to occur during the string conversion process, such as AttributeError for the .lower() method call. Log unexpected exceptions for debugging purposes.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -11,2 +11,7 @@
 -    except Exception as e:
 -        return default
 +    except AttributeError:
 +        return default
 +    except Exception as e:
 +        import logging
 +        logging.error(f"Unexpected error in convert_str_to_bool: {e}")
 +        raise
 +
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged